speech-to-text/recognize_stream.js

```diff
@@ -91,7 +91,7 @@ RecognizeStream.prototype.initialize = function() {
 
   // when the input stops, let the service know that we're done
   self.on('finish', function() {
-    if (self.socket) {
+    if (self.socket && self.socket._readyState === W3CWebSocket.OPEN) {
```

Comment:
It would be preferable to use the `readyState` property (which is exposed https://github.com/theturtle32/WebSocket-Node/blob/master/lib/W3CWebSocket.js#L67) rather than the `_readyState`